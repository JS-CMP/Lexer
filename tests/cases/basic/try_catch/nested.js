try {
    console.log("Outer try");
    try {
        throw "Inner error";
    } catch (e) {
        console.log("Caught inner:", e);
    }
} catch (e) {
    console.log("Caught outer:", e);
}
