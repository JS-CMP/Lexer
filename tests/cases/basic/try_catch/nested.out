#include "types/JsAny.hpp"
#include "global/global.hpp"
#include "types/objects/Types.hpp"
#include "customOperators/CustomOperators.hpp"

int main() {try {console[u"log"](JS::Any(u"Outer try"));try {throw JS::Any(u"Inner error");}catch (JS::Any& e) {console[u"log"](JS::Any(u"Caught inner:"),e);};}catch (JS::Any& e) {console[u"log"](JS::Any(u"Caught outer:"),e);};}
